src/cascadia/TerminalApp/AppKeyBindings.cpp

```diff
@@ -1,117 +1,294 @@
+    DEFINE_EVENT(AppKeyBindings, NewTabWithProfile, _NewTabWithProfileHandlers, TerminalApp::NewTabWithProfileEventArgs);
+    DEFINE_EVENT(AppKeyBindings, NewWindow,         _NewWindowHandlers,         TerminalApp::NewWindowEventArgs);
+    DEFINE_EVENT(AppKeyBindings, CloseWindow,       _CloseWindowHandlers,       TerminalApp::CloseWindowEventArgs);
+    DEFINE_EVENT(AppKeyBindings, CloseTab,          _CloseTabHandlers,          TerminalApp::CloseTabEventArgs);
+    DEFINE_EVENT(AppKeyBindings, SwitchToTab,       _SwitchToTabHandlers,       TerminalApp::SwitchToTabEventArgs);
+    DEFINE_EVENT(AppKeyBindings, NextTab,           _NextTabHandlers,           TerminalApp::NextTabEventArgs);
+    DEFINE_EVENT(AppKeyBindings, PrevTab,           _PrevTabHandlers,           TerminalApp::PrevTabEventArgs);
+    DEFINE_EVENT(AppKeyBindings, IncreaseFontSize,  _IncreaseFontSizeHandlers,  TerminalApp::IncreaseFontSizeEventArgs);
+    DEFINE_EVENT(AppKeyBindings, DecreaseFontSize,  _DecreaseFontSizeHandlers,  TerminalApp::DecreaseFontSizeEventArgs);
+    DEFINE_EVENT(AppKeyBindings, ScrollUp,          _ScrollUpHandlers,          TerminalApp::ScrollUpEventArgs);
+    DEFINE_EVENT(AppKeyBindings, ScrollDown,        _ScrollDownHandlers,        TerminalApp::ScrollDownEventArgs);
+
+
+    // Method Description:
+    // - Deserialize an AppKeyBindings from the key mappings that are in the
+    //   array `json`. The json array should contain an array of objects with
+    //   both a `command` string and a `keys` array, where `command` i one of
+    //   the names listed in `commandNames`, and `keys` is an array of
+    //   keypresses. Currently, the array should contain a single string, which
+    //   can be deserialized into a KeyChord.
+    // Arguments:
+    // - json: and arrayof JsonObject's to deserialize into our _keyShortcuts mapping.
+    // Return Value:
+    // - the newly constructed AppKeyBindings object.
+    TerminalApp::AppKeyBindings AppKeyBindings::FromJson(Windows::Data::Json::JsonArray const& json)
+    {
+        TerminalApp::AppKeyBindings newBindings{};
+
+        for (const auto& value : json)
+        {
+            if (value.ValueType() == JsonValueType::Object)
+            {
+                JsonObject obj = value.GetObjectW();
+                if (obj.HasKey(COMMAND_KEY) && obj.HasKey(KEYS_KEY))
+                {
+                    const auto commandString = obj.GetNamedString(COMMAND_KEY);
+                    const auto keys = obj.GetNamedArray(KEYS_KEY);
+                    if (keys.Size() != 1)
+                    {
+                        continue;
+                    }
+                    const auto keyChordString = keys.GetAt(0).GetString();
+                    ShortcutAction action;
+                    bool parsedSuccessfully = false;
+                    // Try matching the command to one we have
+                    for (const auto& pair : commandNames)

```

Comment:
do we more often look them up by the first or the second? if the second, invert the map to be keyed on binding name

Reply:
yea we weren't ever doing a lookup the way it was